src/SonarScanner.MSBuild.Common/MsBuildPathSettings.cs

```diff
@@ -26,20 +26,48 @@ namespace SonarScanner.MSBuild.Common
 {
     public class MsBuildPathSettings : IMsBuildPathsSettings
     {
+        private readonly Func<Environment.SpecialFolder, bool, string> getEnvironmentSpecialFolderPath;
+        private readonly Func<bool> isWindows;
+
+        public MsBuildPathSettings()
+        {
+            this.getEnvironmentSpecialFolderPath = (folder, forceCreate) =>
+                Environment.GetFolderPath(folder,
+                    forceCreate ? Environment.SpecialFolderOption.Create : Environment.SpecialFolderOption.None);
+            this.isWindows = PlatformHelper.IsWindows;
+        }
+
+        public /* for testing purposes */ MsBuildPathSettings(
```

Comment:
I would have preferred to use `internal` but this was forcing me to add all the extra complexity for the internal visible to.